pyproject.toml

```diff
@@ -9,6 +9,7 @@ requires = [
 [project]
 authors = [{ email = "allen.goodman@icloud.com", name = "Allen Goodman" }]
 dependencies = [
+    "optree",
```

Comment:
At some point we should evaluate whether we need optree or whether the PyTree implementation in PyTorch is sufficient.